go/arrow/compute/internal/kernels/_lib/base_arithmetic_avx2_amd64.s

```diff
@@ -0,0 +1,12469 @@
+	.text
+	.intel_syntax noprefix
+	.file	"base_arithmetic.cc"
+	.globl	arithmetic_avx2                 # -- Begin function arithmetic_avx2
+	.p2align	4, 0x90
+	.type	arithmetic_avx2,@function
+arithmetic_avx2:                        # @arithmetic_avx2
```

Comment:
Looks this file is to generate another assembly file compatible to go, is it strictly necessary to check in this file?

Reply:
It helps make it reproducible and follows the pattern the repo uses for the other cases where we do this. That said, you are correct that it's not strictly necessary to check in this file other than to allow `make` to recognize it doesn't need to rebuild from a clean clone, and without requiring a C++ compiler for consumers unless they run `make clean`.